tests/utils/bulkPerformance.py

```diff
@@ -277,8 +289,174 @@ def generate_value_row(res_value: int, exp_value: int) -> tuple[str, str, str]:
+    }
+  </style>"""
+
+def generate_js_for_search(): 
+    return """
+    function searchTable() {
+      const input = document.getElementById('searchInput');
+      const filter = input.value.toUpperCase();
+      const table = document.getElementById('myTable');
+      const tr = table.getElementsByTagName('tr');
+
+      for (let i = 1; i < tr.length; i++) {
+        let td = tr[i].getElementsByTagName('td');
+        let found = false;
+
+        for (let j = 0; j < td.length; j++) {
+          if (td[j]) {
+            let txtValue = td[j].textContent || td[j].innerText;
+            if (txtValue.toUpperCase().indexOf(filter) > -1) {
+              found = true;
+              break;
+            }
+          }
+        }
+
+        tr[i].style.display = found ? '' : 'none';
+      }
+    }
+    """
+
+
+def generate_bar_chart(mod: MeMoModel):
+  metabs = mod.metabolites
+  keys = dict()
+  for m in metabs:
+      for annotation in m.annotations.keys():
+          keys[annotation] = keys.get(annotation, 0) + 1
+  # Check if order is presercerd in keys/values
+  a = keys.keys()
+  b = keys.values()
+  # TODO CHECK MATH (is the missing calculation correct)
+  # THE MATH IS BS
+  s = """
+       // Get the context of the canvas element
+   var ctx = document.getElementById('myBarChart').getContext('2d');
+    
+   // Create the bar chart
+   var myBarChart = new Chart(ctx, {
+     type: 'bar', // Bar chart type
+     data: {
+         labels: """ + str(list(a)) +  """, // Labels for the bars
+       datasets: [{
+         label: 'Values', // Label for the dataset
+         data: """ + str(list(b)) + """,// Data for the bars
+         backgroundColor: ['#FF5733', '#33FF57', '#3357FF', '#FF33A1'], // Bar colors
+         borderColor: ['#FF5733', '#33FF57', '#3357FF', '#FF33A1'], // Border colors
+         borderWidth: 1
+       }]
+     },
+     options: {
+       scales: {
+         y: {
+           beginAtZero: true, // Ensure the Y-axis starts at zero
+         }
+       },
+       responsive: true // Make the chart responsive
+     }
+     });"""
+  return s
+
+        
+def generate_bac_specs( name: str,  mod: MeMoModel):
+    file_path = os.path.join("bac_specs", f'{name}.html')
+
+    
+    body = """<input type="text" id="searchInput"  onkeyup="searchTable()" placeholder="Search for names..">\n"""
+    body += """<div class="container">\n"""
+    body += bac_spec_content(mod)
+    body += """<canvas id="myBarChart" class = "box" width="200px" height="100px"></canvas>\n"""
+    body += "</div>"
+    body +=  """<script>"""
+    body += generate_js_for_search()
+    body += generate_bar_chart(mod)
+    body +=  """</script>"""
+    html_content = """
+    <!DOCTYPE html>
+    <html lang="en">
+    <head>
+      <meta charset="UTF-8">
+      <meta name="viewport" content="width=device-width, initial-scale=1.0">
+      <script src="https://cdn.jsdelivr.net/npm/chart.js"></script>
```

Comment:
This line requires internet connection? I think it would be better to just download the fire and load it locally